seat.h

```diff
@@ -53,8 +64,6 @@ struct cg_keyboard {
 	struct cg_seat *seat;
 	struct wlr_input_device *device;
 
-	struct wl_listener modifiers;
-	struct wl_listener key;
```

Comment:
I found this in the commit introducing `wlr_keyboard_group`[[1]], is that why?

> Since a wlr_keyboard_group is one logical keyboard, the keys are a set.
This means that if a key is pressed on multiple keyboard devices, the
key event will only be emitted once, but the internal state will count
the number of devices that the key is pressed on. Likewise, the key
release will not be emitted until the key is released from all devices.
If the compositor wants access to which keys are pressed and released
on each keyboard device, the events for those devices can be listened
to, as they currently are, in addition to the group keyboard's events.

[1]: https://github.com/swaywm/wlroots/commit/f2d3b1000f451003c38933c05a742b76da48aeee

Reply:
Yes, since sway does not add a keyboard to a group in some cases (there is configuration as to how sway should handle keyboards), they still need to handle keyboards separately sometimes. This does not apply to cage